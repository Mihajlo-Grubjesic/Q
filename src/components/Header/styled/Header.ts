import styled from 'styled-components';
export const HeaderWrapper = styled.div`
  display: flex;
  justify-content: end;
  padding: 8px;
`;
