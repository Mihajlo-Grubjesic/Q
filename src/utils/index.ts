import { Post, User, Comment, PostData } from "../types";

export const combinePostsData = ({
  posts,
  users,
  comments,
}: {
  posts: Post[];
  users: User[];
  comments: Comment[];
}): PostData[] => {
  return posts.map((post) => {
    const username =
      users.find((user) => user.id === post.userId)?.username ?? "";
    const postComments = comments
      .filter((comment) => comment.postId === post.id)
      .map((comment) => comment.body);
    return { ...post, username, comments: postComments };
  });
};
