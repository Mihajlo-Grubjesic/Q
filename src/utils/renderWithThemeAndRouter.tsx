import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { DefaultTheme, ThemeProvider } from 'styled-components';
import { lightTheme } from '../theme/themes';

export function renderWithThemeAndRouter(
  children: React.ReactElement,
  theme?: DefaultTheme,
) {
  return render(
    <ThemeProvider theme={theme ? theme : lightTheme}>
      <Router>{children}</Router>
    </ThemeProvider>,
  );
}
