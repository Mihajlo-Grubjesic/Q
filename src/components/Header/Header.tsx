import { ThemeToggler } from '../ThemeToggler/ThemeToggler';
import { HeaderWrapper } from './styled/Header';

interface Props {
  onToggleTheme: () => void;
}

export const Header = ({ onToggleTheme }: Props): JSX.Element => {
  return (
    <HeaderWrapper>
      <ThemeToggler onToggleTheme={onToggleTheme} />
    </HeaderWrapper>
  );
};
