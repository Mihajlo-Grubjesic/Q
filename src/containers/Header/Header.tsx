import { ThemeToggler } from './components/ThemeToggler/ThemeToggler';
import { LanguageToggler } from './components/LanguageToggler/LanguageToggler';
import { HeaderWrapper } from './styled/Header';
import { withMessage } from '../../components/HOC/withMessage';

interface Props {
  onToggleTheme: () => void;
  message?: string;
}

const Header = ({ onToggleTheme, message = '' }: Props): JSX.Element => {
  console.log(`${message} Header`);

  return (
    <HeaderWrapper>
      <LanguageToggler />
      <ThemeToggler onToggleTheme={onToggleTheme} />
    </HeaderWrapper>
  );
};

export default withMessage(Header);
