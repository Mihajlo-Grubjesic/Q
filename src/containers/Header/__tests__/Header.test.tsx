import { vi } from 'vitest';
import { renderWithThemeAndRouter } from '../../../utils/renderWithThemeAndRouter';
import Header from '../Header';

describe('Header', () => {
  it('renders header', () => {
    const { container } = renderWithThemeAndRouter(
      <Header onToggleTheme={() => vi.fn()} />,
    );
    expect(container).toMatchSnapshot();
  });
});
