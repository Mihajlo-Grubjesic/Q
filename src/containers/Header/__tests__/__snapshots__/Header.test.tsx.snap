// Vitest Snapshot v1

exports[`Header > renders header 1`] = `
<div>
  <div
    class="sc-gswNZR kBhlT"
  >
    <button
      class="sc-bcXHqe fetPcC"
    >
      🇬🇧 🇪🇸
    </button>
    <button
      class="sc-bcXHqe fetPcC"
      role="theme-toggler"
    >
      ☀️ 🌙
    </button>
  </div>
</div>
`;
